### ISSUE TITLE
[feature request] Expose DOMPresentationUtils APIs

### ISSUE THREAD

**Issue by Andarist** [puppeteer/puppeteer#1188]
I'm interested in having access the APIs from [this](https://chromium.googlesource.com/chromium/blink/+/master/Source/devtools/front_end/components/DOMPresentationUtils.js#289), especially interesting for scraping is `cssPath`.

**Comment by aslushnikov** [comment]
I don't see why it should be a part of puppeteer; you can use devtools source code in your project if you'd like to. 

Btw, you're referring to an old code; a modern version of this file in devtools could be found [here](https://github.com/ChromeDevTools/devtools-frontend/blob/5f320158093624a2759a404177d96d59479aad9d/front_end/components/DOMPresentationUtils.js)

**Comment by Andarist** [comment]
@aslushnikov thanks for the quick reply. 

For starters I cannot use this code as is because [there is no such method as `nodeType`](https://github.com/ChromeDevTools/devtools-frontend/blob/a17fd18e07e88769bb2f45ec1d7a6fed1700d259/front_end/components/DOMPresentationUtils.js#L304) on the `HTMLElement`. Dunno where is it coming from - some custom built-ins extension maybe?

Also it is not packaged, so it introduces problems with versioning and keeping things up to date.

And I have problems with running callback functions in `evaluate`. I guess those functions must be sent as some remote objects to the headless chrome from the node's context and they cannot use references from the node's scope - which makes sense, but makes using external modules cumbersome.

I've ended up using [some fork of mentioned devtools code](https://github.com/maximilianschmitt/cssman/blob/3cebbd8e83fa23c779870aa4b97fbc2e8b3e95cd/main.js), which I had to modify further so the all functions used by `cssPath` are made inline in its body - that was the only way for me (at least I do not know other for now, have just started using `puppeteer`) to make it work, because now the inner functions are getting parsed in the chrome's context and can be used there.

Also the feature was proposed as the convenience for the users, cause I have found this particular thing to be useful for a tool such as `puppeteer`.
